### PR TITLE
Extract databases, elastic pools, and check name availability from sql.core.json

### PR DESCRIPTION
**Pull request by jaredmoo**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
This is a refactoring of existing Microsoft.Sql swagger docs with no functional change. We will be introducing a new version of databases, elastic pools, and check name availability APIs in the near future, so I have extracted the current (soon to be old) versions of these APIs from sql.core.json so that they can be included in readme.md independently.

File: specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/databases.json [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+          },
+          "properties": {
+            "$ref": "#/definitions/DatabaseProperties",
+            "description": "The properties representing the resource.",
+            "x-ms-client-flatten": true
+          }
+        },
+        "allOf": [
+          {
+            "$ref": "#/definitions/TrackedResource"
```
Comment (jhendrixMSFT) [inline specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/databases.json]: This ref does not resolve and is causing CI failures (same for elasticPools.json).

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

**Comment by jhendrixMSFT** [comment]
@jaredmoo I regenerated the Go SDK with these changes on top of the latest SDK for this swagger and there is some content missing, specifically the databaseAutomaticTuning.json and serverAutomaticTuning.json swaggers are no longer referenced in this readme, see the [reference](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md) in head.  Is this expected or is your branch a bit behind?

**Comment by jaredmoo** [comment]
Autotuning was only just merged 1 hour ago so my branch was behind. Now up-to-date :)

**Review by jhendrixMSFT — APPROVED** [review]
I verified that the generated SDK with these changes is identical to what's already in master.

**Pull request merged**